pkg/promclient/promclient.go

```diff
@@ -687,6 +689,49 @@ func (c *Client) BuildVersion(ctx context.Context, base *url.URL) (string, error
 	return b.Data.Version, nil
 }
 
+// LowestTimestamp returns the lowest timestamp in the TSDB by parsing the /metrics endpoint
+// and extracting the prometheus_tsdb_lowest_timestamp_seconds metric from it.
+func (c *Client) LowestTimestamp(ctx context.Context, base *url.URL) (int64, error) {
+	u := *base
+	u.Path = path.Join(u.Path, "/metrics")
```

Comment:
Well, it sucks we can't get this from API.

Reply:
Yeah the approach to read it from /metrics is a bit janky, but i cannot think of another one. That query would probably be prohibitively expensive